src/components/list/list.ts

```diff
@@ -48,13 +57,20 @@ export class MdListItem implements AfterContentInit {
     this._renderer.setElementClass(this._element.nativeElement, 'md-list-avatar', avatar != null);
   }
 
-  constructor(private _renderer: Renderer, private _element: ElementRef) {}
+  constructor(private _renderer: Renderer, private _element: ElementRef) {
+    this._hasLink = _element.nativeElement.tagName.toLowerCase() == 'a';
```

Comment:
Hmm, worried about how this will play in non-browser environments.  It might make more sense to create another class that selects for `a[md-list-item]` specifically and inherits from this one.